middleware/assert_test.go

```diff
@@ -14,106 +14,149 @@ import (
 	"github.com/ONSdigital/dp-filter-api/mock"
 	"github.com/ONSdigital/dp-filter-api/models"
 
-	"github.com/ONSdigital/dp-net/v2/responder"
 	"github.com/ONSdigital/dp-api-clients-go/v2/dataset"
+	"github.com/ONSdigital/dp-net/v2/responder"
 
 	"github.com/gorilla/mux"
 	. "github.com/smartystreets/goconvey/convey"
 )
 
 const (
 	filterBlueprintID = "filter_blueprint_id"
-	testBody = "test body"
-	testToken = "testToken"
+	filterOutputID    = "filter_output_id"
+	testBody          = "test body"
+	testToken         = "testToken"
 )
 
-func TestAssertFilterType(t *testing.T){
-	Convey("Given a healthy dp-cantabular-filter-flex-api", t, func() {
-		expectedResponse := &http.Response{
-			Body:       io.NopCloser(bytes.NewReader([]byte(testBody))),
-			StatusCode: http.StatusOK,
-			Header: map[string][]string{
-				"X-Test": []string{"Value"},
+func setupFilterFlexAPIMock() *apimock.FilterFlexAPIMock {
+	expectedResponse := &http.Response{
+		Body:       io.NopCloser(bytes.NewReader([]byte(testBody))),
+		StatusCode: http.StatusOK,
+		Header: map[string][]string{
+			"X-Test": []string{"Value"},
+		},
+	}
+
+	return &apimock.FilterFlexAPIMock{
+		ForwardRequestFunc: func(r *http.Request) (*http.Response, error) {
+			return expectedResponse, nil
+		},
+	}
+}
+
+func setupDataStoreMock(filterIsFlexible bool, testBluePrintID string) *apimock.DataStoreMock {
+	datastoreMock := mock.NewDataStore().Mock
+	datastoreMock.GetFilterOutputFunc = func(ctx context.Context, filterOutputID string) (*models.Filter, error) {
+		return &models.Filter{
+			Dataset: &models.Dataset{
+				Version: 1,
+			},
+			ID: filterOutputID,
+			Links: models.LinkMap{
+				FilterBlueprint: &models.LinkObject{ID: testBluePrintID, HRef: ""},
 			},
+		}, nil
+	}
+	datastoreMock.GetFilterFunc = func(ctx context.Context, filterID, etag string) (*models.Filter, error) {
+		filter := &models.Filter{
+			Dataset: &models.Dataset{
+				Version: 1,
+			},
+			ID: filterID,
+		}
+		if filterIsFlexible {
+			filter.Type = "flexible"
 		}
+		return filter, nil
+	}
+	return datastoreMock
+}
 
-		filterFlexAPIMock := &apimock.FilterFlexAPIMock{
-			ForwardRequestFunc: func(r *http.Request) (*http.Response, error) {
-				return expectedResponse, nil
-			},
+func setupDatasetAPIMock(datasetIsFlexible bool) *apimock.DatasetAPIMock {
+	datasetAPIMock := &apimock.DatasetAPIMock{}
+
+	datasetAPIMock.GetFunc = func(ctx context.Context, ut, st, cid, dsid string) (dataset.DatasetDetails, error) {
+		dataset := dataset.DatasetDetails{
+			Type: "not_flexible",
 		}
+		if datasetIsFlexible {
+			dataset.Type = "cantabular_flexible_table"
+		}
+		return dataset, nil
+	}
+	return datasetAPIMock
```

Comment:
Can just define these inline if you like:

```go
func setupDatasetAPIMock(datasetIsFlexible bool) *apimock.DatasetAPIMock {
	return &apimock.DatasetAPIMock{
		GetFunc: func(_ context.Context, _, _, _, _ string) (dataset.DatasetDetails, error) {
			details := dataset.DatasetDetails{
				Type: "not_flexible",
			}
			if datasetIsFlexible {
				details.Type = "cantabular_flexible_table"
			}
			return details, nil
		},
	}
}
```

No matter either way.

Reply:
Addressed.